code-name-taran/src/com/spaghettisoft/component/sixtysyx/Player.java

```diff
@@ -0,0 +1,292 @@
+				switch (possibleMarriage[i].getColor())
+				{
+					case SPADES:
+						count1++;
+						continue;
+
+					case HEARTHS:
+						count2++;
+						continue;
+
+					case DIAMONDS:
+						count3++;
+						continue;
+
+					case CLUBS:
+						count4++;
+						continue;
+				}
+			}
+		}
+
+		if (count1 == 2 || count2 == 2 || count3 == 2 || count4 == 2)
+		{
+			return true;
+		}
+
+		return false;
+	}
+
+	private PlayingCard getKinsAndQueens(PlayingCard playingCard)
+	{
+		if (playingCard.getRank().equals(Rank.KING) || playingCard.getRank().equals(Rank.QUEEN))
+		{
+			return playingCard;
+		}
+
+		return null;
+	}
+
+	private boolean isSwitchOptionEnabled()
+	{
+		for (int i = 0; i < hand.length; i++)
+		{
+			if (hand[i].getRank().equals(Rank.NINE) && hand[i].getColor().equals(SixtySix.powerCard.getColor()))
+			{
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public String getName()
+	{
+		return name;
+	}
+
+	/**
+	 * Adds card to the player's hand
+	 */
+	public void addCardtoHand(PlayingCard newCard)
+	{
+		for (int i = 0; i < hand.length; i++)
+		{
+			if (hand[i] == null)
+			{
+				hand[i] = newCard;
+				return;
+			}
+		}
+	}
+
+	public PlayingCard selectMenuItem(int index, boolean isSecond)
+	{
+		if (index < currentHandSize)
+		{
+			PlayingCard temp = hand[index];
+			removeCard(index);
+			currentHandSize--;
+			return temp;
+		}
+		else
+		{
+			if (!isSecond)
+			{
+				AditionalOptions(index);
+			}
+		}
+
+		return null;
+	}
+
+	private void AditionalOptions(int index)
+	{
+		if (marriageInHand())
+		{
+			wonCardsValue += 20;
+			if (isPowerMarriage())
+			{
+				wonCardsValue += 20;
+			}
+			return;
+		}
+
+		switchCards();
+	}
+
+	private void switchCards()
+	{
+		PlayingCard temp = SixtySix.powerCard;
+		for (int i = 0; i < hand.length; i++)
+		{
+			if (hand[i].getRank().equals(Rank.NINE) && hand[i].getColor().equals(SixtySix.powerCard.getColor()))
+			{
+				SixtySix.powerCard = hand[i];
+				hand[i] = temp;
+			}
+		}
+	}
+
+	private void removeCard(int index)
+	{
+		int i = index;
+
+		while (i < hand.length - 1 && hand[i + 1] != null)
+		{
+			PlayingCard temp = hand[i];
+			hand[i] = hand[i + 1];
+			hand[i + 1] = temp;
+			i++;
+		}
+		hand[i] = null;
+
+	}
+
+	private boolean isPowerMarriage()
+	{
+		boolean queen = false;
+		boolean king = false;
+
+		for (int i = 0; i < hand.length; i++)
+		{
+			if (hand[i].getRank().equals(Rank.QUEEN) && hand[i].getColor().equals(SixtySix.powerCard.getColor()))
+			{
+				queen = true;
+			}
+			if (hand[i].getRank().equals(Rank.KING) && hand[i].getColor().equals(SixtySix.powerCard.getColor()))
+			{
+				queen = true;
+			}
+		}
+		return queen && king;
+	}
+
+	public void setWonHand(PlayingCard first, PlayingCard second)
+	{
+		wonCardsValue += first.getRank().value() + second.getRank().value();
+	}
+
+	public int getWonCardsValue()
+	{
+		return wonCardsValue;
+	}
+
+	public void orderHand()
+	{
+		sortByColor();
+		sortByRank();
+	}
+
+	private void sortByColor()

```

Comment:
I think the Collection.sort(), could save you some time in writing the algorithm. Same for the other method : )